_posts/2021-10-01-debezium-server-iceberg.adoc

```diff
@@ -0,0 +1,163 @@
+
+Note: For the tables without record key(PK) operation mode falls back to `append` even configuration is set to upsert mode.
+
+==== Keeping Deleted Records
+
+For some use cases it's useful to keep deleted records as soft deletes, this is possible by setting `debezium.sink.iceberg.upsert-keep-deletes` to `true`.
+This setting will keep the latest version of deleted record (`__op=d`) in the iceberg table. Setting it to false will remove deleted records from the destination table.
+
+==== Append mode
+
+This is most straightforward operation mode, setting `debezium.sink.iceberg.upsert` to `false` sets the operation mode to append.
+With append mode data deduplication is not done and all received records are appended to destination table.
+
+=== Optimizing batch size (commit interval)
+
+Debezium extracts/consumes database events in real time and this could cause too frequent commits( generate too many small files) to iceberg table,
+which is not optimal for batch processing especially when near realtime data feed is sufficient.
+To avoid this problem it's possible to use following configuration and increase batch size per commit.
+
+**MaxBatchSizeWait**: uses debezium metrics to optimize batch size, it periodically reads streaming queue current size and waits until it reaches to `max.batch.size`.
+Maximum wait and check intervals are controlled by `debezium.sink.batch.batch-size-wait.max-wait-ms`, `debezium.sink.batch.batch-size-wait.wait-interval-ms` properties.
+During the wait debezium events are collected in memory (in debezium streaming queue), this way each commit receives more records and consistent batch size.
+Note: this setting should be configured together with `debezium.source.max.queue.size` and `debezium.source.max.batch.size` debezium properties
+
+Note: It's also possible to do data compaction using iceberg, compacting data and metadata files to get the best performance.
+
+example setting::
+
+[source,properties]
+----
+debezium.sink.batch.batch-size-wait=MaxBatchSizeWait
+debezium.sink.batch.metrics.snapshot-mbean=debezium.postgres:type=connector-metrics,context=snapshot,server=testc
+debezium.sink.batch.metrics.streaming-mbean=debezium.postgres:type=connector-metrics,context=streaming,server=testc
+debezium.source.connector.class=io.debezium.connector.postgresql.PostgresConnector
+debezium.source.max.batch.size=50000
+debezium.source.max.queue.size=400000
+debezium.sink.batch.batch-size-wait.max-wait-ms=60000
+debezium.sink.batch.batch-size-wait.wait-interval-ms=10000
+----
+
+=== Destination, iceberg catalog
```

Comment:
Having a section title for a single sentence seems a bit too fine-grained. It is a general question though which is a bit unclear to me: how are the files actually stored in Iceberg? Using some object storage like S3? Can I store it in my local filesystem too? Can we have an example for this somehow?